### PR TITLE
added flag to skip adding initContainers to the OpenSearch StatefulSet

### PR DESCRIPTION
**Pull request by djcass44**
Signed-off-by: Django Cass <django@dcas.dev>

This Operator is unable to be used on OpenShift or clusters that block pods from running as `root` as the `initContainer` used to `chown` the data volume is hardcoded to run as `root` and cannot be disabled.

This MR adds a flag to the OpenSearch CustomResourceDefinition which allows users to skip the addition of the `chown` `initContainer`. The flag is named `skip` rather than `enabled` to retain backwards compatibility with previous versions of the Operator.

I personally don't consider removing the `initContainer` a loss of functionality as the `chown` behaviour is provided by modern CSI-based Kubernetes clusters via the [`fsGroup` security context functionality](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods).

Relevant issue: #197 - this MR should allow OpenShift users to run OpenSearch **without the `anyuid` SCC**

### CODE REVIEW

**Comment by swoehrl-mw** [comment]
Hi @djcass44. 
I can see and agree that making the chown initcontainer optional is worthwile. But is the CRD the best place to configure this? Looking at it this is behaviour that is the same for all opensearch clusters in the same kubernetes cluster (the kubernetes cluster either is openshift or has the csi capability or not). So IMO this is something that should be better configured in the helm chart values and not in the CRD.

**Comment by djcass44** [comment]
@swoehrl-mw that's a good point. I've updated the MR to use an environment variable and modified the chart to allow the user to set it.

File: charts/opensearch-operator/Chart.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 2.2.0
+version: 2.2.1
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/Chart.yaml:18]: No need to update the version, our release process will do that automatically.

File: opensearch-operator/pkg/builders/cluster.go [hunk 2]
```diff
@@ -221,8 +222,9 @@ func NewSTSForNodePool(
 	}
 	mainCommand := helpers.BuildMainCommand("./bin/opensearch-plugin", cr.Spec.General.PluginsList, true, startUpCommand)
 
-	initContainers := []corev1.Container{
-		{
+	var initContainers []corev1.Container
+	if os.Getenv(helpers.SkipInitContainerEnvVariable) != "true" {
```
Comment (swoehrl-mw) [inline opensearch-operator/pkg/builders/cluster.go:226]: There is `strconv.ParseBool` to parse a string to a bool. That allows the users to use different variants of `true`.
Also please move the env lookup part into `helpers/constants.go` (like already done with `ParallelRecoveryMode` and `ClusterDnsBase`).

File: opensearch-operator/pkg/builders/cluster_test.go [hunk 3]
```diff
@@ -54,6 +54,18 @@ func ClusterDescWithAdditionalConfigs(addtitionalConfig map[string]string, boots
 var _ = Describe("Builders", func() {
 
 	When("Constructing a STS for a NodePool", func() {
+		It("should include the init containers", func() {
+			var clusterObject = ClusterDescWithVersion("2.2.1")
+			var result = NewSTSForNodePool("foobar", &clusterObject, opsterv1.NodePool{}, "foobar", nil, nil, nil)
+			Expect(len(result.Spec.Template.Spec.InitContainers)).To(Equal(1))
+		})
+		It("should skip the init containers", func() {
```
Comment (swoehrl-mw) [inline opensearch-operator/pkg/builders/cluster_test.go:62]: Please extend the description a bit to make clear what is tested. Ala "should skip the init container as SKIP_INIT_CONTAINER is set".

File: opensearch-operator/pkg/builders/cluster_test.go [hunk 4]
```diff
@@ -54,6 +54,18 @@ func ClusterDescWithAdditionalConfigs(addtitionalConfig map[string]string, boots
...
+			_ = os.Setenv(helpers.SkipInitContainerEnvVariable, "true")
+			var clusterObject = ClusterDescWithVersion("2.2.1")
+			var result = NewSTSForNodePool("foobar", &clusterObject, opsterv1.NodePool{}, "foobar", nil, nil, nil)
+			Expect(len(result.Spec.Template.Spec.InitContainers)).To(Equal(0))
+			_ = os.Unsetenv(helpers.SkipInitContainerEnvVariable)
+		})
```
Comment (swoehrl-mw) [inline opensearch-operator/pkg/builders/cluster_test.go:68]: Please also test the `NewBootstrapPod` function in the same way.

File: charts/opensearch-operator/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Added
 - Added support for custom image used by `kubeRbacProxy`.
+- Added support for setting arbitrary environment variables on the operator-manager
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/CHANGELOG.md:11]: No need to add this, we now only use the changelog in the gitub release.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
Left some comments. Please also resolve the merge conflicts.

**Comment by djcass44** [comment]
@swoehrl-mw thanks for the comments. I believe I have addressed them all.

**Review by swoehrl-mw — APPROVED** [review]
LGTM

**Pull request merged**